### PR TITLE
Components and pages

### PR DESCRIPTION
**Pull request by TusharJade**
Added: Navbar, sidebar, video listing page with basic category filtering, added video context

### CODE REVIEW

**Review by SumitSaundadkar — APPROVED** [review]
code looks clean to me you just added components and pages now go for functionality.

File: src/pages/SingleVideoPage/SingleVideoPage.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  const { video } = useVideoContext();
+  const { videoId } = useParams();
+
+  const singleVideoFunc = (videoData, id) => {
+    return videoData.find((item) => item._id === id);
+  };
+
+  const singleVideoData = singleVideoFunc(video, videoId);
+
+  console.log(singleVideoData);
```
Comment (Avin008) [inline src/pages/SingleVideoPage/SingleVideoPage.jsx:15]: remove console.log

**Review by Avin008 — APPROVED** [review]
clean && well-structured code. good to merge

File: src/pages/SingleVideoPage/SingleVideoPage.jsx (hunk 1 above)
Comment (TusharJade) [reply src/pages/SingleVideoPage/SingleVideoPage.jsx:15]: yup will remove it

**Pull request merged**